src/systems/physics/Physics.cc

```diff
@@ -212,6 +213,12 @@ class ignition::gazebo::systems::PhysicsPrivate
   /// \brief Keep track of what entities are static (models and links).
   public: std::unordered_set<Entity> staticEntities;
 
+  /// \brief Keep track of non-static link poses.
+  /// This allows for skipping pose updates if a link's pose didn't change
+  /// after a physics step.
+  public: std::unordered_map<Entity, ignition::physics::Pose<double, 3>>
```

Comment:
I hardcoded `double` and `3` because these are the values that seem to match the template values of the link's world pose in `UpdateSim` (I ran `gdb` with a breakpoint on this line and then ran `whatis worldPose`, which returned `type = const Eigen::Transform<double, 3, 1, 0> &`):
https://github.com/ignitionrobotics/ign-gazebo/blob/86e6ad93581362eba4e89a58dc1fef058806a081/src/systems/physics/Physics.cc#L1727

Is there a better way to define these template values, especially if `double` and `3` aren't always guaranteed to work? From my understanding, `double` is the pose's `Scalar` value, and `3` is the pose's `dimension` value (more information about these template parameters can be found [here](https://github.com/ignitionrobotics/ign-physics/blob/6aa968042659ad4ee60e9126a7ff81fe68d0d05c/include/ignition/physics/Geometry.hh#L43-L44)).

Reply:
This physics system is specifically for `double` scalar and `3` dimension (i.e, `FeaturePolicy3d`), so I think it's okay to hard code them.

Reply:
https://github.com/ignitionrobotics/ign-gazebo/pull/669/commits/6bf05727f38212c16f9b4ac3e26b6d5058f005d8 replaces `ignition::physics::Pose<double, 3>` with `ignition::math::Pose3d`, so I'm marking this as resolved.